client/my-sites/preview/main.jsx

```diff
@@ -66,23 +70,52 @@ class PreviewMain extends React.Component {
 	}
 
 	render() {
-		const { translate } = this.props;
+		const { translate, isPreviewable, site } = this.props;
+
+		if ( ! site ) {
+			// todo: some loading state?
+			return <span></span>;
```

Comment:
You are right, thanks for digging. For some reason when I tried returning `null` last time, I got the error "A valid react component must be returned" or similar so without much testing, I just opted for quick `<span>` fix.